src/Effects/Effect.gd

```diff
@@ -0,0 +1,15 @@
+extends Node
+
+# The idea here is that every effect will be able to process every type of affect-able stat
+# with this class defining stubs that do nothing, so an effect only defines what it needs.
+
+var _parent
+
+func _super(parent):
```

Comment:
Not really happy about this name but it at least clearly communicates intention and behaviour. Oh, you Python nerds :)